### PR TITLE
Bug 1174892 - Gaiatest needs to be updated for treeherder client changes

### PR DESCRIPTION
**Pull request by mwargers**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=49ae045fb1aabbd9027b97809b0fe93368c90d7a

File: tests/python/gaia-ui-tests/gaiatest/mixins/treeherder.py [hunk 1]
```diff
@@ -246,15 +246,12 @@ def post_to_treeherder(self, tests):
...
-            protocol=url.scheme,
-            host=url.hostname,
-            project=project,
-            oauth_key=os.environ.get('TREEHERDER_KEY'),
-            oauth_secret=os.environ.get('TREEHERDER_SECRET'))
+        client = TreeherderClient(protocol=url.scheme, host=url.hostname)
         self.logger.debug('Sending results to Treeherder: %s' %
                           job_collection.to_json())
-        response = request.post(job_collection)
+        response = client.post_collection(project,
```
Comment (wlach) [inline tests/python/gaia-ui-tests/gaiatest/mixins/treeherder.py]: `post_collection` doesn't provide a return value anymore, I'd probably just suggest just calling it and letting it throw an exception if there's an error. Other than that, this looks fine. You'll also want to bump the required version of treeherder-client, of course.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=174705bd28902c6dc8174f3f62c1d21765d8a7b6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3a231ed4826be996af2e9e559088af47d67f83bd

**Pull request merged**